examples/general/non_local_effect_before_error_return/src/lib.rs

```diff
@@ -150,22 +150,21 @@ impl<'tcx> LateLintPass<'tcx> for NonLocalEffectBeforeErrorReturn {
             |path, contributing_calls, span| {
                 // smoelius: The path is from a return to the start block.
                 for (i, &index) in path.iter().enumerate() {
-                    let basic_block = &mir.basic_blocks[index];
-
                     if_chain! {
                         if !contributing_calls.contains(index);
-                        if let Some(call_span) = is_call_with_mut_ref(cx, mir, &path[i..]);
+                        if let Some(func_and_span) = is_call_with_mut_ref(cx, mir, &path[i..]);
```

Comment:
```suggestion
                        if let Some((func, span)) = is_call_with_mut_ref(cx, mir, &path[i..]);
```
Style nit (saves having to write `.0` and `.1`). Note that the code in the block would need to be adjusted.